backend-peer/src/controllers/messages/index.ts

```diff
@@ -0,0 +1,91 @@
+import { Controller, Post, Route, SuccessResponse, Tags, Body, Get } from 'tsoa'
+import { injectable } from 'tsyringe'
+
+import { CloudagentManager } from '../../lib/services/cloudagent.js'
+
+import { logger } from '../../lib/logger.js'
+import { Message } from '../types.js'
+import { ConnectionRecord, DidExchangeState } from '@aries-framework/core'
+
+const log = logger.child({ context: 'MessagesController' })
+
+@Route('api/messages')
+@Tags('messages')
+@injectable()
+export class MessagesController extends Controller {
+  constructor(private cloudagent: CloudagentManager) {
+    super()
+  }
+
+  /**
+   * @summary Send a new message.
+   * @description Check if connections exist if not, create one, then send message.
+   */
+  @SuccessResponse(204)
+  @Post('/')
+  public async post(@Body() body: { did: string; message: Message }) {
+    log.debug({
+      msg: 'new request received',
+      controller: '/messages',
+      payload: body,
+    })
+    try {
+      const { did, message } = body
+      const connections = await this.cloudagent.getConnections()
+      let connectionId: string | undefined
+      let connectionRecord: ConnectionRecord | undefined
+
+      for (const { id, theirDid, state } of connections) {
+        if (theirDid === did && state == 'completed') {
+          connectionId = id
+        }
+      }
+      if (connectionId === undefined) {
+        //create a connection if one does not exist
+        connectionRecord = await this.cloudagent.receiveImplicitInvitation(
+          did,
+          true
+        )
+        if (!connectionRecord) {
+          throw new Error('Failed to create connection record') // Handle the case where connectionRecord is undefined
+        }
+        connectionId = connectionRecord.id
+      }
+
+      return this.cloudagent.sendMessage(connectionId, message)
+    } catch (e) {
+      throw new Error(`${e}`) //Remove this before pushing as we do not want to propagate errors to users
+    }
+  }
+
+  /**
+   * @summary Get all messages.
+   * @description Get all open connections and all messages and
+   */
+  @SuccessResponse(200)
+  @Get('/')
+  public async get() {
+    log.info({ msg: 'Getting all messages', controller: '/messages' })
+    const allMessages: Message[] = []
+    try {
+      const params = { state: DidExchangeState.Completed }
+      const connections = await this.cloudagent.getConnections(params)
+      for (const { id, theirDid } of connections) {
+        const messagesperId = await this.cloudagent.getMessages(id)
+        for (const { content, role } of messagesperId) {
+          if (role == 'sender') {
+            allMessages.push({ content: content, recipientDid: theirDid })
+          } else if (role == 'receiver') {
+            allMessages.push({ content: content, senderDid: theirDid })
+          } else {
+            allMessages.push({ content: content, senderDid: 'none found' }) //this ok?
+          }
+        }
+      }
```

Comment:
could use `map()`s so you don't have to make an array and push to it

Reply:
I could what would you want a s the map structure? 
should messages be mapped to a connection? or to sender/receiver did?